stonesoup/predictor/kalman.py

```diff
@@ -376,3 +393,78 @@ def predict(self, prior, timestamp=None, **kwargs):
 
         # and return a Gaussian state based on these parameters
         return GaussianStatePrediction(x_pred, p_pred, timestamp=timestamp)
+
+
+class SqrtKalmanPredictor(KalmanPredictor):
+    r"""The version of the Kalman predictor that operates on the square root parameterisation of
+    the Gaussian state, :class:`~.SqrtGaussianState`.
+
+    The prediction is undertaken in one of two ways. The default is to work in exactly the same
+    way as the parent class, with the exception exception that the predicted covariance is
+    subject to a Cholesky factorisation prior to initialisation of the :class:`~.SqrtGaussianState`
+    output. The alternative, accessible via the :attr:`qr_method = True` flag, is to predict via a
+    modified Gram-Schmidt process. See [1] for details.
+
+    If transition and control models are possessed of the square root form of the covariance (as
+    :attr:`sqrt_covar` in the case of the transition model and :attr:`sqrt_control_noi` for
+    control models), then these are used directly. If not then they are created from the full
+    matrices using the scipy.linalg :meth:`sqrtm()` method. (Unlike the Cholesky decomposition
+    this works on positive semi-definite matrices, as well as positive definite ones.
+
+    References
+    ----------
+    1. Maybeck, P.S. 1994, Stochastic Models, Estimation, and Control, Vol. 1, NavtechGPS,
+       Springfield, VA.
+
+    """
+    qr_method = Property(bool, default=False, doc="A switch to do the prediction via a QR "
+                                                  "decomposition, rather than using a Cholesky"
+                                                  "decomposition.")
+
+    _prediction_class = SqrtGaussianStatePrediction
+
+    def _predicted_covariance(self, prior, predict_over_interval, **kwargs):
+        """Private function to return the predicted covariance.
+
+        Parameters
+        ----------
+        prior : :class:`~.SqrtGaussianState`
+            The prior class (which carries the covariance in square root form)
+        predict_over_interval : :class`~.timedelta`
+            The interval over which the model is applied
+
+        Returns
+        -------
+        : :class:`~.CovarianceMatrix`
+            The predicted covariance matrix
+
+        """
+        sqrt_prior_cov = prior.sqrt_covar
+
+        trans_m = self._transition_matrix(prior=prior, time_interval=predict_over_interval,
+                                          **kwargs)
+        if hasattr(self.transition_model, 'sqrt_covar'):
+            sqrt_trans_cov = self.transition_model.sqrt_covar(time_interval=predict_over_interval,
+                                                              **kwargs)
+        else:
+            sqrt_trans_cov = la.sqrtm(self.transition_model.covar(
+                time_interval=predict_over_interval, **kwargs))
```

Comment:
Dunno. Do you mean that we'll only have to change one instance of `sqrt_covar` if that's to be renamed?

Reply:
Ah, I thought I had deleted that last line. I suppose it doesn't make a massive difference, but yeah that was my initial though.

I also remember that in the pre-beta era we were strongly advocating the use of EAFP (i.e. try-catch) over type checking (isinstance, hasattr, etc.), but not too sure were we stand on this now, as a lot of the code that did EAFP has been phased out. I thought I'd mention it just in case.

Reply:
Yeah, EAFP is usually preferred style in Python, so should go with that. Also better fits with duck typing approach in Python.

Reply:
Ok, hereby asking for forgiveness. (4 times as far as I could see.)